test/unit/models/schema_spec.js

```diff
@@ -0,0 +1,154 @@
+"use strict";
+
+var Instance = require("../../../lib/models/services-schema");
+var _        = require("lodash");
+var expect   = require("chai").expect;
+
+describe("The services schema", function () {
+
+	var VALID_ID            = "somestringwith12903";
+	var VALID_TYPE          = "t2.micro";
+	var VALID_AMI           = "ami-d05e75b8";
+	var VALID_STATE_PENDING = "pending";
+	var VALID_STATE_READY   = "ready";
+	var VALID_URI           = "https://location.com";
+
+	var INVALID_AMI         = [ "ami-imagename", "ami-imagename-2" ];
+	var INVALID_STATE       = "something else";
+	var INVALID_URI         = "this is not a uri";
+
+	function describeInitWith (description, options, fails) {
+		var expectedOptions = _.defaults(
+			options,
+			{
+				id    : undefined,
+				type  : "t2.micro",
+				ami   : "ami-d05e75b8",
+				state : "pending",
+				uri   : null
+			}
+		);
+		describe("initialized with " + description, function () {
+			if (!fails) {
+				it("returns a services object", function () {
+					var instances = new Instance(options);
+					expect(instances, "properties").to.deep.equal(expectedOptions);
+				});
+			}
+			else {
+				it("fails", function () {
+					expect(function () { return new Instance(options); }, "throws").to.throw(Error);
+				});
+			}
+		});
+	}
+
+	describeInitWith("all valid parameters",
+		{
+			id    : VALID_ID,
+			type  : VALID_TYPE,
+			ami   : VALID_AMI,
+			state : VALID_STATE_READY,
+			uri   : VALID_URI
+		}
+	);
+
+	describeInitWith("all valid parameters",
```

Comment:
It's probably better to not duplicate descriptions. This one and the previous one could probably be more specific about what they are testing (the two different valid states).